content/v2/oauth.markdown

```diff
@@ -16,15 +16,63 @@ Developers need to register their application before getting started. A register
+
+<info>
+The API currently supports only the OAuth 2 [Authorization Code Grant](http://tools.ietf.org/html/rfc6749#section-4.1) flow. Therefore, [the only supported value for the `response_type` is `code`](http://tools.ietf.org/html/rfc6749#section-3.1.1).
+</info>
 
 ### Step 1 - Authorization
 
+This starts the OAuth 2.0 authorization flow. This isn't an API call — it's the web page that lets the user sign in to DNSimple and authorize your app. After the user decides whether or not to authorize your app, they will be redirected to the URI specified by `redirect_uri`.
+
+Your web or mobile app should redirect users to the following URL:
+
+~~~
+GET https://dnsimple.com/oauth/authorize
+~~~
+
+The following values should be passed as GET parameters:
+
+`response_type` | **Required**. The grant type requested. We currently only support `code`.
+`client_id`     | **Required**. The client ID you received from DNSimple when you registered the application.
+`redirect_uri`  | Where to redirect the user after authorization has completed. This must be the exact URI registered or a subdirectory.
+`scope`         | We currently don't use this field.
+`state`         | An unguessable random string. It is used to protect against cross-site request forgery attacks and it will be passed back to your redirect URI.
+
+Because `/oauth/authorize` is a website, there is no direct return value. However, after the user authorizes your app, they will be sent to your redirect URI.
+
+If the request is successful, these parameters are passed in the query string in the URL:
+
+`code`          | The authorization code, which can be used in the next step to retrieve a bearer token.
+`state`         | The state content, if any, originally passed to `/oauth/authorize`.
+
+If an error occurs, including if the user has chosen not to authorize the app, the following parameters will be included in the redirect URI:
+
+`error`             | An error code per [Section 4.1.2.1 of the OAuth 2.0 spec](http://tools.ietf.org/html/rfc6749#section-4.1.2.1).
+`error_description` | A user-friendly description of the error that occurred.
+`state`             | The state content, if any, originally passed to `/oauth/authorize`.
+
 ### Step 2 - Access Token
 
+This API method is used to exchange the `code` with a bearer token you can use to authenticate to the DNSimple API.
+
+~~~
+POST https://dnsimple.com/oauth/access_token
+~~~
+
+The following values should be passed as POST parameters:
+
+`client_id`     | **Required**. The client ID you received from DNSimple when you registered the application.
+`client_secret` | **Required**. The client secret you received from DNSimple when you registered the application.
+`code`          | **Required**. The code acquired in the previous authorization step.
+`redirect_uri`  | Only used to validate that it matches the original `/oauth/authorize`, not used to redirect again.
```

Comment:
Good question, I need to check. At this point the state should be useless because you already have a one-time `code` (which guarantees against forged requests).